### PR TITLE
Add pyproject.toml

### PR DESCRIPTION
**Pull request by GernotMaier**
This PR adds a `pyproject.toml` file following the [CTA project-template-python-pure ](https://github.com/cta-observatory/project-template-python-pure) guidelines. This allows to provide a distribution of gammasim-tools with executables.

To try the installation out do e.g.,:
```
mamba create --name simtools-prod python=3.9
mamba activate simtools-prod
pip install .
```

Applications can be called now directly from the command line:
```
simtools-print-array-elements -h
```
Note the consistent naming starting with `simtools-` and the replacement of `_` by `-`. Applications which are not in the main directory are expected to be for developers and not installed (at this point this is valid for those in applications/db_development_tools).

Most important changes:
- added pyproject.toml
- moved `applications` folder to `simtools/applications`

Addresses issues #424, #423, #422

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  { name="Victor Barbosa Martins", email="victor.barbosa.martins@desy.de" },
+  { name="Gernot Maier", email="gernot.maier@desy.de" },
+]
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: BSD-3-Clause",
+    "Operating System :: OS Independent",
+]
+requires-python = ">=3.9"
+dependencies = [
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply pyproject.toml:22]: It doesn't include the tools needed for development (e.g., pytest) and to generate documentation (e.g., sphinx). They are listed later, but are not needed for the general user.

This is btw not a nice thing: we still need an environment.yml file and a pyproject.toml file.

**Comment by VictorBarbosaMartins** [comment]
I tried the installation steps and it worked out to run the application as `print_array_elements -h`
I think this implementation is really helpful for using the applications in the future, from my side you can go ahead

**Comment by VictorBarbosaMartins** [comment]
Hi @GernotMaier , I just saw a typo in the name of the environment: `mamba activate gammsim-tools-prod` should be `mamba activate gammasim-tools-prod`

File: environment.yml [hunk 2]
```diff
@@ -8,7 +8,6 @@ dependencies:
   - numpy
   - astropy
   - matplotlib-base
-  - sphinx
```
Comment (orelgueta) [inline environment.yml:11]: Is it intentional to remove sphinx?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+[project.urls]
+"documentation" = "https://gammasim.github.io/simtools/"
+"repository" = "https://github.com/gammasim/simtools"
+"bug tracker" = "https://github.com/gammasim/simtools/issues"
+
+[project.scripts]
+simtools-add-file-to-db = "simtools.applications.add_file_to_db:main"
+simtools-compare-cumulative-psf = "simtools.applications.compare_cumulative_psf:main"
+simtools-db-development-tools = "simtools.applications.db_development_tools:main"
```
Comment (orelgueta) [inline pyproject.toml:57]: This is not an application

**Review by orelgueta — COMMENTED** [review]
One small change is necessary. I still couldn't test it cause my conda has gone haywire, but it's probably not a big deal and we can merge anyway.

File: environment.yml (hunk 2 above)
Comment (GernotMaier) [reply environment.yml:11]: sphinx is still there, it was twice in the environment file.

File: pyproject.toml (hunk 3 above)
Comment (GernotMaier) [reply pyproject.toml]: Yes, it is a directory. I've removed it.

**Review by orelgueta — APPROVED** [review]
OK, looks good now, thanks!

**Pull request merged**